From b37cf7358dd27c8c3b96e0787907aac481a1b006 Mon Sep 17 00:00:00 2001
From: Ray Chiang <rchiang@apache.org>
Date: Tue, 5 Dec 2017 10:05:14 -0800
Subject: [PATCH 2707/2848] Revert "YARN-7363. ContainerLocalizer don't have a
 valid log4j config in case of Linux container
 executor. (Contributed by Yufei Gu)"

This reverts commit 60bd43360d9a1f6bee697ef7ca29aca628e08799.

Change-Id: Ib3daffa2c61b8a8fb05d16d599269994e3fc5160
---
 .../apache/hadoop/yarn/conf/YarnConfiguration.java |    6 ---
 .../src/main/resources/yarn-default.xml            |    8 ----
 .../server/nodemanager/LinuxContainerExecutor.java |   27 +-----------
 .../WindowsSecureContainerExecutor.java            |    2 +-
 .../localizer/ContainerLocalizer.java              |   46 +-------------------
 .../TestLinuxContainerExecutorWithMocks.java       |   19 +++-----
 6 files changed, 11 insertions(+), 97 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java
index e849eb8..96433a8 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java
@@ -1062,12 +1062,6 @@ private static void addDeprecatedKeys() {
   public static final String NM_CONTAINER_LOCALIZER_JAVA_OPTS_DEFAULT =
       "-Xmx256m";
 
-  /** The log level of container localizer process. */
-  public static final String NM_CONTAINER_LOCALIZER_LOG_LEVEL=
-      NM_PREFIX + "container-localizer.log.level";
-  public static final String NM_CONTAINER_LOCALIZER_LOG_LEVEL_DEFAULT =
-      "INFO";
-
   /** The Docker image name(For DockerContainerExecutor).*/
   public static final String NM_DOCKER_CONTAINER_EXECUTOR_IMAGE_NAME =
     NM_PREFIX + "docker-container-executor.image-name";
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml
index b86028c..7e23734 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml
@@ -841,14 +841,6 @@
 
   <property>
     <description>
-      The log level for container localizer while it is an independent process.
-    </description>
-    <name>yarn.nodemanager.container-localizer.log.level</name>
-    <value>INFO</value>
-  </property>
-
-  <property>
-    <description>
       Where to store container logs. An application's localized log directory
       will be found in ${yarn.nodemanager.log-dirs}/application_${appid}.
       Individual containers' log directories will be below this, in directories 
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java
index 07264b5..3130a20 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java
@@ -43,7 +43,6 @@
 import org.apache.hadoop.yarn.conf.YarnConfiguration;
 import org.apache.hadoop.yarn.server.nodemanager.containermanager.container.Container;
 import org.apache.hadoop.yarn.server.nodemanager.containermanager.container.ContainerDiagnosticsUpdateEvent;
-import org.apache.hadoop.yarn.server.nodemanager.containermanager.launcher.ContainerLaunch;
 import org.apache.hadoop.yarn.server.nodemanager.containermanager.linux.privileged.PrivilegedOperation;
 import org.apache.hadoop.yarn.server.nodemanager.containermanager.linux.privileged.PrivilegedOperationException;
 import org.apache.hadoop.yarn.server.nodemanager.containermanager.linux.privileged.PrivilegedOperationExecutor;
@@ -251,9 +250,6 @@ public void startLocalizer(Path nmPrivateContainerTokensPath,
     }
     command.addAll(ContainerLocalizer.getJavaOpts(getConf()));
     buildMainArgs(command, user, appId, locId, nmAddr, localDirs);
-
-    Path containerLogDir = getContainerLogDir(dirsHandler, appId, locId);
-    command = replaceWithContainerLogDir(command, containerLogDir);
     String[] commandArray = command.toArray(new String[command.size()]);
     ShellCommandExecutor shExec = new ShellCommandExecutor(commandArray,
         null, null, 0L, true);
@@ -275,32 +271,11 @@ public void startLocalizer(Path nmPrivateContainerTokensPath,
     }
   }
 
-  private List<String> replaceWithContainerLogDir(List<String> commands,
-      Path containerLogDir) {
-    List<String> newCmds = new ArrayList<>(commands.size());
-
-    for (String item : commands) {
-      newCmds.add(item.replace(ApplicationConstants.LOG_DIR_EXPANSION_VAR,
-          containerLogDir.toString()));
-    }
-
-    return newCmds;
-  }
-
-  private Path getContainerLogDir(LocalDirsHandlerService dirsHandler,
-      String appId, String containerId) throws IOException {
-    String relativeContainerLogDir = ContainerLaunch
-        .getRelativeContainerLogDir(appId, containerId);
-
-    return dirsHandler.getLogPathForWrite(relativeContainerLogDir,
-        false);
-  }
-
   @VisibleForTesting
   public void buildMainArgs(List<String> command, String user, String appId,
       String locId, InetSocketAddress nmAddr, List<String> localDirs) {
     ContainerLocalizer.buildMainArgs(command, user, appId, locId, nmAddr,
-        localDirs, super.getConf());
+      localDirs);
   }
 
   @Override
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/WindowsSecureContainerExecutor.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/WindowsSecureContainerExecutor.java
index ce8101d..4b2ea21 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/WindowsSecureContainerExecutor.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/WindowsSecureContainerExecutor.java
@@ -699,7 +699,7 @@ public void startLocalizer(Path nmPrivateContainerTokens,
      command.addAll(ContainerLocalizer.getJavaOpts(getConf()));
 
      ContainerLocalizer.buildMainArgs(command, user, appId, locId, nmAddr, 
-         localDirs, super.getConf());
+         localDirs);
      
      String cmdLine = StringUtils.join(command, " ");
      
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ContainerLocalizer.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ContainerLocalizer.java
index 55d5da0..f82f894 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ContainerLocalizer.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ContainerLocalizer.java
@@ -53,8 +53,6 @@
 import org.apache.hadoop.security.token.TokenIdentifier;
 import org.apache.hadoop.util.DiskChecker;
 import org.apache.hadoop.yarn.YarnUncaughtExceptionHandler;
-import org.apache.hadoop.yarn.exceptions.YarnRuntimeException;
-import org.apache.hadoop.yarn.api.ApplicationConstants;
 import org.apache.hadoop.yarn.api.records.LocalResource;
 import org.apache.hadoop.yarn.api.records.SerializedException;
 import org.apache.hadoop.yarn.conf.YarnConfiguration;
@@ -338,12 +336,8 @@ private LocalizerStatus createStatus() throws InterruptedException {
    */
   public static void buildMainArgs(List<String> command,
       String user, String appId, String locId,
-      InetSocketAddress nmAddr, List<String> localDirs, Configuration conf) {
-
-    String logLevel = conf.get(YarnConfiguration.
-            NM_CONTAINER_LOCALIZER_LOG_LEVEL,
-        YarnConfiguration.NM_CONTAINER_LOCALIZER_LOG_LEVEL_DEFAULT);
-    addLog4jSystemProperties(logLevel, command);
+      InetSocketAddress nmAddr, List<String> localDirs) {
+    
     command.add(ContainerLocalizer.class.getName());
     command.add(user);
     command.add(appId);
@@ -355,16 +349,6 @@ public static void buildMainArgs(List<String> command,
     }
   }
 
-  private static void addLog4jSystemProperties(
-      String logLevel, List<String> command) {
-    command.add("-Dlog4j.configuration=container-log4j.properties");
-    command.add("-D" + YarnConfiguration.YARN_APP_CONTAINER_LOG_DIR + "=" +
-        ApplicationConstants.LOG_DIR_EXPANSION_VAR);
-    command.add("-D" + YarnConfiguration.YARN_APP_CONTAINER_LOG_SIZE + "=0");
-    command.add("-Dhadoop.root.logger=" + logLevel + ",CLA");
-    command.add("-Dhadoop.root.logfile=container-localizer-syslog");
-  }
-
   public static void main(String[] argv) throws Throwable {
     Thread.setDefaultUncaughtExceptionHandler(new YarnUncaughtExceptionHandler());
     // usage: $0 user appId locId host port app_log_dir user_dir [user_dir]*
@@ -374,7 +358,6 @@ public static void main(String[] argv) throws Throwable {
     // MKDIR $x/$user/appcache/$appid/filecache
     // LOAD $x/$user/appcache/$appid/appTokens
     try {
-      createLogDir();
       String user = argv[0];
       String appId = argv[1];
       String locId = argv[2];
@@ -410,31 +393,6 @@ public static void main(String[] argv) throws Throwable {
     }
   }
 
-  /**
-   * Create the log directory, if the directory exists, make sure its permission
-   * is 750.
-   */
-  private static void createLogDir() {
-    FileContext localFs;
-    try {
-      localFs = FileContext.getLocalFSFileContext(new Configuration());
-
-      String logDir = System.getProperty(
-          YarnConfiguration.YARN_APP_CONTAINER_LOG_DIR);
-
-      if (logDir != null && !logDir.trim().isEmpty()) {
-        Path containerLogPath = new Path(logDir);
-        FsPermission containerLogDirPerm= new FsPermission((short)0750);
-        localFs.mkdir(containerLogPath, containerLogDirPerm, true);
-        // set permission again to make sure the permission is correct
-        // in case the directory is already there.
-        localFs.setPermission(containerLogPath, containerLogDirPerm);
-      }
-    } catch (IOException e) {
-      throw new YarnRuntimeException("Unable to create the log dir", e);
-    }
-  }
-
   private static void initDirs(Configuration conf, String user, String appId,
       FileContext lfs, List<Path> localDirs) throws IOException {
     if (null == localDirs || 0 == localDirs.size()) {
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
index 8815077..aebb12a 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorWithMocks.java
@@ -189,7 +189,7 @@ public void testStartLocalizer() throws IOException {
     try {
       mockExec.startLocalizer(nmPrivateCTokensPath, address, "test", "application_0", "12345", dirsHandler);
       List<String> result=readMockParams();
-      Assert.assertEquals(result.size(), 23);
+      Assert.assertEquals(result.size(), 18);
       Assert.assertEquals(result.get(0), YarnConfiguration.DEFAULT_NM_NONSECURE_MODE_LOCAL_USER);
       Assert.assertEquals(result.get(1), "test");
       Assert.assertEquals(result.get(2), "0" );
@@ -197,17 +197,12 @@ public void testStartLocalizer() throws IOException {
       Assert.assertEquals(result.get(4), "/bin/nmPrivateCTokensPath");
       Assert.assertEquals(result.get(8), "-classpath" );
       Assert.assertEquals(result.get(11), "-Xmx256m" );
-      Assert.assertEquals(result.get(12), "-Dlog4j.configuration=container-log4j.properties" );
-      Assert.assertEquals(result.get(13), "-Dyarn.app.container.log.dir=${yarn.log.dir}/userlogs/application_0/12345");
-      Assert.assertEquals(result.get(14), "-Dyarn.app.container.log.filesize=0");
-      Assert.assertEquals(result.get(15), "-Dhadoop.root.logger=INFO,CLA");
-      Assert.assertEquals(result.get(16), "-Dhadoop.root.logfile=container-localizer-syslog");
-      Assert.assertEquals(result.get(17),"org.apache.hadoop.yarn.server.nodemanager.containermanager.localizer.ContainerLocalizer" );
-      Assert.assertEquals(result.get(18), "test");
-      Assert.assertEquals(result.get(19), "application_0");
-      Assert.assertEquals(result.get(20),"12345" );
-      Assert.assertEquals(result.get(21),"localhost" );
-      Assert.assertEquals(result.get(22),"8040" );
+      Assert.assertEquals(result.get(12),"org.apache.hadoop.yarn.server.nodemanager.containermanager.localizer.ContainerLocalizer" );
+      Assert.assertEquals(result.get(13), "test");
+      Assert.assertEquals(result.get(14), "application_0");
+      Assert.assertEquals(result.get(15),"12345" );
+      Assert.assertEquals(result.get(16),"localhost" );
+      Assert.assertEquals(result.get(17),"8040" );
 
     } catch (InterruptedException e) {
       LOG.error("Error:"+e.getMessage(),e);
-- 
1.7.9.5

